daras_ai_v2/base.py

```diff
@@ -490,8 +486,17 @@ def _render_unpublished_changes_indicator(self):
             gui.html("Unpublished changes")
 
     def _render_saved_timestamp(self, pr: PublishedRun):
+        last_run_at = gui.session_state.get(
+            StateKeys.updated_at, datetime.datetime.today()
+        )
+
+        if isinstance(last_run_at, str):
+            last_run_at = datetime.datetime.fromisoformat(last_run_at)
+
         with gui.tag("span", className="text-muted"):
-            gui.write(f"{get_relative_time(pr.updated_at)}")
+            gui.write(
+                f"{get_relative_time(pr.updated_at if pr.is_root() else last_run_at)}"
```

Comment:
you wanna show the published run last updated_at time on the published run root, and show the saved run updated_at time on a saved run